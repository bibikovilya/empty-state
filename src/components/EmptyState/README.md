Wrapping component for handle empty state.

Props:
- data: array of data need to check;
- fallback: override component for empty state.

Default props:
- data: []
- fallback: `<EmptyComponent />`

```
<EmptyState data={[...]} fallback={<OverrideEmptyComponent />}>
  ...
</EmptyState>
```
