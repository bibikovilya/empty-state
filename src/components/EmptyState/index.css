.card {
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 300px;
  width: 300px;
  background-color: #f5f7f7;
  color: #88a2a6;
  text-align: center;
  border-radius: 10px;
  padding: 10px;
  font-family: Helvetica;
  font-size: 14px;
}

.card__logo {
  margin: 0 auto;
}
