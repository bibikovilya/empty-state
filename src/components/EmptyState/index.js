import React from 'react';
import PropTypes from 'prop-types';

const EmptyState = ({ data, children }) => (
  data.length === 0
    ? <span>Empty</span>
    : children
)

EmptyState.propTypes = {
  data: PropTypes.arrayOf(PropTypes.any),
}

export default EmptyState;
