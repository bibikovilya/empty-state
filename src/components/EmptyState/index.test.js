import React from 'react';
import TestRenderer from 'react-test-renderer';

import EmptyState from '.';

describe('EmptyState', () => {
  it('should render correctly without props', () => {
    const component = TestRenderer.create(
      <EmptyState />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
