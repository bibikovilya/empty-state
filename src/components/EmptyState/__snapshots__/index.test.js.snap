// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState should render correctly without props 1`] = `
<div
  className="card"
>
  <p>
    We don't have what to show you
  </p>
  <img
    alt="No data"
    className="card__logo"
    src="chatchamp-logo.png"
  />
</div>
`;
